core/trino-main/src/main/java/io/trino/operator/DirectExchangeClientStatus.java

```diff
@@ -162,6 +162,47 @@ public DirectExchangeClientStatus mergeWith(DirectExchangeClientStatus other)
                 requestDuration.mergeWith(other.requestDuration)); // this is correct as long as all clients have the same shape of histogram
     }
 
+    @Override
+    public DirectExchangeClientStatus mergeWith(List<DirectExchangeClientStatus> others)
+    {
+        if (others.isEmpty()) {
+            return this;
+        }
+
+        long bufferedBytes = this.bufferedBytes;
+        long maxBufferedBytes = this.maxBufferedBytes;
+        long averageBytesPerRequest = this.averageBytesPerRequest;
+        long successfulRequestsCount = this.successfulRequestsCount;
+        int bufferedPages = this.bufferedPages;
+        int spilledPages = this.spilledPages;
+        long spilledBytes = this.spilledBytes;
+        boolean noMoreLocations = this.noMoreLocations;
+        ImmutableList.Builder<TDigestHistogram> requestDurations = ImmutableList.builderWithExpectedSize(others.size());
+        for (DirectExchangeClientStatus other : others) {
+            bufferedBytes = (bufferedBytes + other.bufferedBytes) / 2; // this is correct as long as all clients have the same buffer size (capacity)
```

Comment:
If both values are the same, do we need to do something with them?
cannot just do:
```java
bufferedBytes = this.bufferedBytes;
```
outside loop?